mygpo/api/legacy.py

```diff
@@ -21,42 +21,97 @@
 
 @never_cache
 @csrf_exempt
-def upload(request):
+def upload(request, branch_coverage):
     try:
+        # Branch ID: 0
+        branch_coverage[0] = True
         emailaddr = request.POST["username"]
         password = request.POST["password"]
         action = request.POST["action"]
         protocol = request.POST["protocol"]
         opml = request.FILES["opml"].read()
     except MultiValueDictKeyError:
+        # Branch ID: 1
+        branch_coverage[1] = True
         return HttpResponse("@PROTOERROR", content_type="text/plain")
 
     user = auth(emailaddr, password)
     if not user:
+        # Branch ID: 2
+        branch_coverage[2] = True
         return HttpResponse("@AUTHFAIL", content_type="text/plain")
+    else:
+        # Branch ID: 3
+        branch_coverage[3] = True
 
     dev = get_device(user, LEGACY_DEVICE_UID, request.META.get("HTTP_USER_AGENT", ""))
 
-    existing_urls = [x.url for x in dev.get_subscribed_podcasts()]
+    # existing_urls = [x.url for x in dev.get_subscribed_podcasts()]
+
+    # Break down the iteration
+    existing_urls = []
+    for x in dev.get_subscribed_podcasts():
+        # Branch ID: 3
+        branch_coverage[3] = True
+        existing_urls.append(x['url'])
+        
 
     i = Importer(opml)
 
-    podcast_urls = [p["url"] for p in i.items]
+    # podcast_urls = [p["url"] for p in i.items]
+
+    # Break down iteration
+    podcast_urls = []
+    for p in i.items:
+        # Branch ID: 4
```

Comment:
Missing the coverage[4] = true statement

Reply:
Thats why this was never triggering, will fix